app/assets/javascripts/application.js

```diff
@@ -154,7 +154,10 @@ function check_caps_lock(key, e) {
 
 function remove_fields(link) {
   $(link).prev("input[type=hidden]").val("1");
-  $(link).closest(".fields").hide();
+  var $field_row = $(link).closest(".fields"),
+  $error_block = $field_row.prev("tr.error-msg-block");
+  if ($error_block) { $error_block.hide(); }
```

Comment:
Nitpick: no need for the if, calling hide on an empty error_block won't cause an error.